ibmcloud/image/build.sh

```diff
@@ -143,21 +162,53 @@ done < <(parted -s -m "$src_nbd" unit B print)
 sleep 1
 resize2fs -f "${src_nbd}p1"
 
-case "$disklabel" in
-    gpt)
-        sgdisk "$src_nbd" -R "$tmp_nbd"
-        sgdisk -G "$tmp_nbd"
-        ;;
-    msdos)
-        sfdisk -d "$src_nbd" | sfdisk "$tmp_nbd"
-        ;;
-esac
+if [ "${SE_BOOT}" = "1" ]; then
+    echo "Creating boot-se and root partitions" 
+    parted -a optimal $tmp_nbd mklabel gpt \
+        mkpart boot-se ext4 1MiB 256MiB \
+        mkpart root 256MiB "${disksize}" \
+        set 1 boot on
+else
+    case "$disklabel" in
+        gpt)
+            sgdisk "$src_nbd" -R "$tmp_nbd"
+            sgdisk -G "$tmp_nbd"
+            ;;
+        msdos)
+            sfdisk -d "$src_nbd" | sfdisk "$tmp_nbd"
+            ;;
+    esac
+fi
 
-for part_number in "${parts[@]}"; do
-    dd if="${src_nbd}p$part_number" of="${tmp_nbd}p$part_number" bs=$((1024*1024))
-done
+if [ "${SE_BOOT}" = "1" ]; then
+    echo "Waiting for the two nbd partitions to show up"
+    while true; do
+    sleep 1
+    [ -e ${tmp_nbd}p2 ] && break
+    done
+    echo -e "\nFormatting boot-se partition"
+    mke2fs -t ext4 -L boot-se ${tmp_nbd}p1
+    export BOOT_UUID=$(blkid ${tmp_nbd}p1 -s PARTUUID -o value)
+    export ROOT_UUID=$(blkid ${tmp_nbd}p2 -s PARTUUID -o value)
+    echo UUIDs:
+    echo "Boot UUID=$BOOT_UUID"
+    echo "Root UUID=$ROOT_UUID"
+    echo -e "\nSetting up encrypted root partition"
+    dd if=/dev/random of=rootkey.bin bs=1 count=64 &> /dev/null
+    cat rootkey.bin | \
```

Comment:
nit: Any reason why you do a `cat` and do not pass `rootkey.bin` as argument to `--keyfile` directly?